File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.go
func longestConsecutive(nums []int) int {
longest := 0
hmp := make(map[int]struct{})
for _, num:= range nums {
	hmp[num] = struct{}{}
}

for _, num := range nums{
	if _, ok := hmp[num-1]; !ok{
		length := 1
		for {
			if _, exist := hmp[num+length]; exist{
				length += 1
			}else{
				break
			}
		}
		if length > longest {
			longest = length
		}
	}
}
return longest
}

File: Data Structures & Algorithms/first-missing-positive/submission-2.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        i,n=0,len(nums)

        while i<n:
            if nums[i] <= 0 or nums[i] > n:
                i+=1
                continue
            index = nums[i]-1
            if nums[i] != nums[index]:
                nums[i],nums[index] = nums[index],nums[i]
            else:
                i+=1
        for i in range(len(nums)):
            if nums[i]-1 != i:
                return i+1
        
        return n+1